src/EventSubscriber/AuditSubscriber.php

```diff
@@ -346,6 +359,11 @@ protected function audit(EntityManager $em, array $data)
             $data[$field] = $meta->idGenerator->generate($em, null);
         }
 
+        // Log the ip address and User Agent String.
+        $mainRequest = $this->requestStack->getMainRequest();
+        $data['ip'] = $this->logIp && $mainRequest ? $mainRequest->getClientIp() : null;
+        $data['userAgent'] = $this->logUserAgent && $mainRequest ? substr($mainRequest->headers->get('User-Agent'), 0, $this->truncateUserAgent) : null;
```

Comment:
I believe the default for the parameter bag is null, and I believe in newer versions of php, you can't pass null to a string function without a warning. Either set the default to a blank string, or cast it to a string.

Reply:
Updated 👍🏼 

have you considered adding codesniffer (ecs)/ or static analysis tool like phpstan to dev dependencies? useful for catching things like this ! possibly useful too if you want to add it to the github workflows too ? I'd be willing to do it in another PR.